tensorflow/core/platform/default/env.cc

```diff
@@ -295,6 +299,12 @@ void PosixEnv::GetLocalTempDirectories(std::vector<string>* list) {
       return;
     }
   }
+  LOG(WARNING) << "We are not able to find a directory for temporary files.\n"
+               << "Verify the directory access and available space under: "
+               << paths << ". "
```

Comment:
Done. But it is a much bigger change. Personally, I prefer the old version. But tell me which one you prefer, I can remove the new version.

If I use `absl::StrJoin(candidates, ",");`, it segfault.
So that function doesn't like the current candidates data structure.
If I change that data structure to std::vector<std::string>

Then I can't make it const with a simple list initializer as getenv() sometimes return NULL and this isn't a valid element.

So I changed the init to use a lambda to initialize it correctly.

Reply:
Sorry for the delay. I agree this is a bigger change. I'm not sure why you're getting segfaults with `absl::StrJoin(candidates, ",");`. My main issue with the old version is that `paths += std::string(d);` results in O(n^2) runtime (probably not important for this particular case, but could theoretically get bad if there are a lot of candidates), and it's generally better to use StrJoin than to implement it yourself. Can you try going back to the old version, but build up a `std::vector<std::string> paths` instead of just a `std::string`, and then `absl::StrJoin(paths, ",")`?

Reply:
@skye done.